### PR TITLE
Fix default value not consistent with interface

### PR DESCRIPTION
**Pull request by CarlSchwan**
```php
<?php
// lib/public/GroupInterface.php:99
public function getGroups($search = '', $limit = -1, $offset = 0);
```

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1: assuming you tested it

File: lib/private/Group/Database.php [hunk 1]
```diff
@@ -280,9 +280,11 @@ public function getGroups($search = '', $limit = null, $offset = null) {
 			)));
 		}
 
-		$query->setMaxResults($limit)
-			->setFirstResult($offset);
-		$result = $query->execute();
+		if ($limit !== -1) {
```
Comment (ArtificialOwl) [inline lib/private/Group/Database.php:283]: setMaxResult() accept negative value ? if no, ($limit > -1) ?

Comment (CarlSchwan) [reply lib/private/Group/Database.php]: done

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
https://github.com/nextcloud/server/pull/36592

**Pull request closed without merging**